archivist/appidp.py

```diff
@@ -22,15 +22,20 @@
 """
 
 from __future__ import annotations
+
 from logging import getLogger
+from typing import TYPE_CHECKING
```

Comment:
`TYPE_CHECKING`  const is only needed if  `from __future__ import annotations` isn't present

Reply:
hmm - not according to the ruff linter

Reply:
I think it's the opinion of the Ruff authors. How type annotations behave is clearly stated in https://peps.python.org/pep-0649/ 

I'll give the point that it clarifies the intention of the imports. But the main problem I have with `TYPE_CHECKING` is that it's not part of the python spec; it's a value that is always False, but linter authors and IDE writers semantically recognise it and agree to ignore it when doing their static analysis.  

This was a few years ago, but I had difficulty getting the pycharm IDE to respect that value.